lms/djangoapps/grades/new/course_grade.py

```diff
@@ -0,0 +1,205 @@
+        for chapter_key in self.course_structure.get_children(
+                self.course.location):
+            chapter = self.course_structure[chapter_key]
+            subsection_grades = []
+            for subsection_key in self.course_structure.get_children(
+                    chapter_key):
+                subsection_grades.append(
+                    subsection_grade_factory.create(
+                        self.course_structure[subsection_key],
+                        self.course_structure, self.course
+                    )
+                )
+
+            self.chapter_grades.append({
+                'display_name': block_metadata_utils.display_name_with_default_escaped(
+                    chapter),
+                'url_name': block_metadata_utils.url_name_for_block(chapter),
+                'sections': subsection_grades
+            })
+
+        self._signal_listeners_when_grade_computed()
+
+    def score_for_module(self, location):
+        """
+        Calculate the aggregate weighted score for any location in the course.
+        This method returns a tuple containing (earned_score, possible_score).
+
+        If the location is of 'problem' type, this method will return the
+        possible and earned scores for that problem. If the location refers to a
+        composite module (a vertical or section ) the scores will be the sums of
+        all scored problems that are children of the chosen location.
+        """
+        if location in self.locations_to_scores:
+            score = self.locations_to_scores[location]
+            return score.earned, score.possible
+        children = self.course_structure.get_children(location)
+        earned = 0.0
+        possible = 0.0
+        for child in children:
+            child_earned, child_possible = self.score_for_module(child)
+            earned += child_earned
+            possible += child_possible
+        return earned, possible
+
+    def _compute_letter_grade(self, percentage):
+        """
+        Returns a letter grade as defined in grading_policy (e.g. 'A' 'B' 'C' for 6.002x) or None.
+
+        Arguments
+        - grade_cutoffs is a dictionary mapping a grade to the lowest
+            possible percentage to earn that grade.
+        - percentage is the final percent across all problems in a course
+        """
+
+        letter_grade = None
+        grade_cutoffs = self.course.grade_cutoffs
+
+        # Possible grades, sorted in descending order of score
+        descending_grades = sorted(grade_cutoffs, key=lambda x: grade_cutoffs[x], reverse=True)
+        for possible_grade in descending_grades:
+            if percentage >= grade_cutoffs[possible_grade]:
+                letter_grade = possible_grade
+                break
+
+        return letter_grade
+
+    def _signal_listeners_when_grade_computed(self):
+        """
+        Signal all listeners when grades are computed.
+        """
+        responses = GRADES_UPDATED.send_robust(
```

Comment:
I can see both sides of this. It is a bit confusing to appear as a constant, but I can also see where indicating the global nature of the variable could be helpful.

If you feel strongly about this feel free to change it, but I think I'm alright with the ALL_CAPS